src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-perf/csvjsonconvertor.py

```diff
@@ -11,6 +11,42 @@
 
 log = helix.logs.get_logger()
 
+# read a row of csv data into dict
+def add_row(test, value, csvdict):
+    """
+        test names are of the form
+        System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        where the string before the first '(' are the identifiers for the test and the string after is function metadata
+        strip off the metadata when generating the namespace/function identifiers and combine them when again for function identifier
+        eg.: System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        will be converted to following identifiers
+        [System, ComponentModel, Tests, Perf_TypeDescriptorTests, GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))]
+    """
+
+    funcMeta = test.split('(')
+    funcMeta = funcMeta[1:]
+    test = test.split('(')[0]
+    identifiers = test.split('.')
+    funcName = identifiers[len(identifiers) - 1]
```

Comment:
You can just write "identifiers[-1]" here. Python has negative indexing for cases like this.